spinalcordtoolbox/image.py

```diff
@@ -1836,3 +1836,36 @@ def generate_stitched_qc_images(ims_in: Sequence[Image], im_out: Image) -> Tuple
     assert im_concat.data.shape == im_out.data.shape
 
     return im_concat, im_out
+
+
+def check_image_kind(img):
+    """
+    Identify the image as one of 4 image types:
+
+        - 'seg': Binary segmentation (0/1)
+        - 'softseg': Nonbinary segmentation in the range [0, 1], where 0 and 1 are the majority of values
+        - 'labeled_seg': Nonbinary, whole values (0, 1, 2...) or (0.0, 1.0, 2.0...) only
+        - 'im': Any other float-valued image where 0 and 1 are not the majority of values.
+
+    Useful for determining A) which colormap should be applied to an image, or
+                           B) which interpolation can be safely used on a given image
+
+    Output strings should be compatible with display_viewer_syntax() function.
+    """
+    # Check if image is a segmentation (binary or soft) by making sure:
+    # - 0/1 are the two most common voxel values
+    # - 0/1 account for >95% of voxels (to allow for some soft voxels)
+    unique, counts = np.unique(np.round(img.data, decimals=1), return_counts=True)
+    unique, counts = unique[np.argsort(counts)[::-1]], counts[np.argsort(counts)[::-1]]  # Sort by counts
+    binary_most_common = set(unique[0:2].astype(float)) == {0.0, 1.0}
+    binary_percentage = np.sum(counts[0:2]) / np.sum(counts)
+    is_whole_only = np.equal(np.mod(unique, 1), 0).all()
+    if binary_most_common:
+        if binary_percentage == 1.0:
+            return 'seg'
+        elif binary_percentage > 0.95:
+            return 'softseg'
+    if is_whole_only:
+        return 'seg-labeled'
+    else:
+        return 'anat'
```

Comment:
> maybe it would be more appropriate to use another colormap (e.g., random) to clearly show different rootlet levels in different colours.

@valosekj to answer your concern about `fsleyes` colours, I've added a new function that will determine the colormap to use based on the image type. (This also results in `red-yellow` being used for `sct_deepseg` softsegs now, rather than just `red`.)

To do this, I re-used some code that I originally developed for determining image kind (to decide on the kind of interpolation used for QC reports). 

However, this code was already a little suspect, as it uses a heuristic to determine image kind. I know @mguaypaq wasn't the biggest fan of this code already, so I'm tagging him to see if he might have alternative suggestions.

Reply:
Side note: As I was working on https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4346, I thought about how it would be might to have the above function for testing purposes, too.)

(For example, if we're expect a binary segmentation, but receive a softseg instead, then this logic could maybe help detect that case? Likewise, if we're expecting integer labels, but get back non-integer labels, it could help detect that case too.)

Reply:
Thanks for the tag. Since this function is used purely for the purpose of suggesting a visualization to the user, I think it's reasonable to do auto-detection like this. There's not really any risk of misleading them.